packages/pricing/src/factory/index.ts

```diff
@@ -0,0 +1,56 @@
+import { ILogger } from "@grants-stack-indexer/shared";
+
+import {
+    CoingeckoProvider,
+    DummyPricingProvider,
+    InvalidPricingSource,
+    IPricingProvider,
+    MissingDependencies,
+    PricingConfig,
+    PricingProvider,
+} from "../internal.js";
+
+/**
+ * Factory class for creating pricing providers.
+ */
+export class PricingProviderFactory {
+    /**
+     * Creates a pricing provider based on the provided configuration.
+     * @param options - The pricing configuration.
+     * @param deps - dependencies to inject into the pricing provider.
+     * @returns The created pricing provider.
+     * @throws {InvalidPricingSource} if the pricing source is invalid.
+     * @throws {MissingDependencies} if the dependencies are missing.
+     */
+    static create(
+        options: PricingConfig<PricingProvider>,
+        deps?: {
+            logger?: ILogger;
+        },
+    ): IPricingProvider {
+        let pricingProvider: IPricingProvider;
+
+        switch (options.pricingSource) {
+            case "dummy":
+                pricingProvider = new DummyPricingProvider(options.dummyPrice);
+                break;
+            case "coingecko":
+                if (!deps?.logger) {
+                    throw new MissingDependencies();
+                }
```

Comment:
I'm wondering if we should standardize the dependencies validation with some kind of generic pattern (with the help of zod maybe?) while initializing objects.

Reply:
you mean inside constructors or having a zod schema here instead of ` if (!deps?.logger)`

Reply:
Zod schema could be a solution yeah. I'm thinking in general, whenever a service needs dependencies and those dependencies might need to accomodate themselves to some particular schema (like optional dependencies, some dependencies need to be present based on some other dependency value, etc)

I guess that applying a zod schema could cover 99% of the cases, I'll apply it in my code and I'll let you know how happy/sad it is to use it lol